stock_request/tests/test_stock_request.py

```diff
@@ -0,0 +1,1237 @@
+        )
+        self.assertEqual(1, len(picking))
+        picking.action_assign()
+        self.assertEqual(3, len(picking.move_line_ids))
+        line = picking.move_line_ids.filtered(lambda r: r.product_id == self.product)
+        line.qty_done = 4
+        line = picking.move_line_ids.filtered(lambda r: r.product_id == product2)
+        line.qty_done = 1
+        self.env["stock.backorder.confirmation"].with_context(
+            button_validate_picking_ids=[picking.id]
+        ).create({"pick_ids": [(4, picking.id)]}).process_cancel_backorder()
+        sr1.env.invalidate_all()
+        sr2.env.invalidate_all()
+        sr3.env.invalidate_all()
+        self.assertEqual(sr1.state, "done")
+        self.assertEqual(sr1.qty_cancelled, 0)
+        self.assertEqual(sr2.state, "cancel")
+        self.assertEqual(sr2.qty_cancelled, 4)
+        self.assertEqual(sr3.state, "cancel")
+        self.assertEqual(sr3.qty_cancelled, 5)
+
+
+class TestStockRequestOrderState(TestStockRequest):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.product_a = cls._create_product(
+            "CODEA",
+            "Product A",
+            cls.main_company.id,
+        )
+        cls.product_a.route_ids = [(6, 0, cls.route.ids)]
+        cls.product_b = cls._create_product(
+            "CODEB",
+            "Product B",
+            cls.main_company.id,
+        )
+        cls.product_b.route_ids = [(6, 0, cls.route.ids)]
+        expected_date = fields.Datetime.now()
+        vals = {
+            "company_id": cls.main_company.id,
+            "warehouse_id": cls.warehouse.id,
+            "location_id": cls.warehouse.lot_stock_id.id,
+            "expected_date": expected_date,
+            "stock_request_ids": [
+                (
+                    0,
+                    0,
+                    {
+                        "product_id": cls.product_a.id,
+                        "product_uom_id": cls.product_a.uom_id.id,
+                        "product_uom_qty": 1.0,
+                        "company_id": cls.main_company.id,
+                        "warehouse_id": cls.warehouse.id,
+                        "location_id": cls.warehouse.lot_stock_id.id,
+                        "expected_date": expected_date,
+                    },
+                ),
+                (
+                    0,
+                    0,
+                    {
+                        "product_id": cls.product_b.id,
+                        "product_uom_id": cls.product_b.uom_id.id,
+                        "product_uom_qty": 1.0,
+                        "company_id": cls.main_company.id,
+                        "warehouse_id": cls.warehouse.id,
+                        "location_id": cls.warehouse.lot_stock_id.id,
+                        "expected_date": expected_date,
+                    },
+                ),
+            ],
+        }
+        cls.order = cls.request_order.new(vals)
```

Comment:
```suggestion
        cls.order = cls.request_order.create(vals)
```

Reply:
@victoralmau Sometimes, create is not necessary, especially in tests to improve performances.